### PR TITLE
Support decom of the 32-bit HDR3 quadrant offsets

### PR DESCRIPTION
**Pull request by taldcroft**
## Description

This provides functionality to decom the 32-bit header-3 quadrant readout offsets.

Most of the new code relates to an unfortunate corner case for quadrant C because the 4 bytes are split across two slots. Most of this code and the tests were generated with Claude Sonnet 4.6.

### Unit tests

- [x] Linux
```
ska3-kady$ git rev-parse --short HEAD
273e01a
ska3-kady$ pytest
=========================================== test session starts ===========================================
platform linux -- Python 3.13.11, pytest-9.0.2, pluggy-1.6.0
rootdir: /data/baffin/tom/git
configfile: pytest.ini
plugins: anyio-4.12.1, timeout-2.4.0
collected 125 items                                                                                       

mica/archive/tests/test_aca_dark_cal.py ..................                                          [ 14%]
mica/archive/tests/test_aca_hdr3.py ..........                                                      [ 22%]
mica/archive/tests/test_aca_l0.py .....                                                             [ 26%]
mica/archive/tests/test_asp_l1.py .......                                                           [ 32%]
mica/archive/tests/test_cda.py ..................................................                   [ 72%]
mica/archive/tests/test_obspar.py .                                                                 [ 72%]
mica/report/tests/test_report.py ss                                                                 [ 74%]
mica/report/tests/test_write_report.py s                                                            [ 75%]
mica/starcheck/tests/test_catalog_fetches.py ...............                                        [ 87%]
mica/stats/tests/test_acq_stats.py ...                                                              [ 89%]
mica/stats/tests/test_guide_stats.py ....                                                           [ 92%]
mica/vv/tests/test_vv.py .........                                                                  [100%]

============================================ warnings summary =============================================
```

Independent check of unit tests by Jean
- [x] Linux
```
jeanconn-kady> pytest
================================================= test session starts ==================================================
platform linux -- Python 3.13.11, pytest-9.0.2, pluggy-1.6.0
rootdir: /proj/sot/ska/jeanproj/git
configfile: pytest.ini
plugins: anyio-4.12.1, timeout-2.4.0
collected 125 items                                                                                                    

mica/archive/tests/test_aca_dark_cal.py ..................                                                       [ 14%]
mica/archive/tests/test_aca_hdr3.py ..........                                                                   [ 22%]
mica/archive/tests/test_aca_l0.py .....                                                                          [ 26%]
mica/archive/tests/test_asp_l1.py .......                                                                        [ 32%]
mica/archive/tests/test_cda.py ..................................................                                [ 72%]
mica/archive/tests/test_obspar.py .                                                                              [ 72%]
mica/report/tests/test_report.py ..                                                                              [ 74%]
mica/report/tests/test_write_report.py .                                                                         [ 75%]
mica/starcheck/tests/test_catalog_fetches.py ...............                                                     [ 87%]
mica/stats/tests/test_acq_stats.py ...                                                                           [ 89%]
mica/stats/tests/test_guide_stats.py ....                                                                        [ 92%]
mica/vv/tests/test_vv.py .........                                                                               [100%]

=================================================== warnings summary ===================================================
mica/mica/archive/tests/test_asp_l1.py::test_update_l1_archive
  /proj/sot/ska3/aca/lib/python3.13/pty.py:95: DeprecationWarning: This process (pid=2859707) is multi-threaded, use of forkpty() may lead to deadlocks in the child.
    pid, fd = os.forkpty()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
====================================== 125 passed, 1 warning in 378.45s (0:06:18) ======================================
jeanconn-kady> git rev-parse HEAD
19b9ab232af8287ef8cc8cdd0525eb8c43364d80

```
### Functional tests

Some functional testing in (unpublished) notebooks related to post-patch pixel telemetry analysis.

### CODE REVIEW

File: mica/archive/tests/test_aca_hdr3.py [hunk 1]
```diff
@@ -55,6 +59,11 @@ def test_MSIDset():
         "zero_off16_quad_b": 10710,
         "zero_off16_quad_c": 10710,
         "zero_off16_quad_d": 10267,
+        "zero_off32_quad_a": 10267,
+        "zero_off32_quad_b": 10267,
+        "zero_off32_quad_c_lsb": 2,
+        "zero_off32_quad_c_msb": 2,
+        "zero_off32_quad_d": 10731,
```
[the remark this answers is not in the document]
Comment (taldcroft) [reply mica/archive/tests/test_aca_hdr3.py:66]: Fixed.

File: mica/archive/aca_hdr3.py [hunk 2]
```diff
@@ -15,10 +15,48 @@
...
+
+    Returns
+    -------
+    idx_lsb : np.ndarray
+        Indices into ``times_lsb`` for each matched pair.
+    idx_msb : np.ndarray
+        Indices into ``times_msb`` for each matched pair.
+    """
+    idx_lsb = np.searchsorted(times_lsb, times_msb, side="left")
+    idx_lsb = np.clip(idx_lsb, 0, len(times_lsb) - 1)
```
[the remark this answers is not in the document]
Comment (taldcroft) [reply mica/archive/aca_hdr3.py:48]: Fixed.

File: mica/archive/aca_hdr3.py [hunk 3]
```diff
@@ -48,15 +84,93 @@ def func(slot_data) -> np.ndarray:
...
+    -------
+    callable
+        Function that takes slot_data and returns an array of raw uint8 byte values.
+    """
+
+    def func(slot_data) -> np.ndarray:
+        m = len(byte_msids)
+        bytes_list = [slot_data[byte_msids[ii]].astype(np.uint8) for ii in range(m)]
+        return bytes_list
+
```
[the remark this answers is not in the document]
Comment (taldcroft) [reply mica/archive/aca_hdr3.py:120]: Fixed.

**Comment by jeanconn** [comment]
For this one, I see I'm tagged for review.  I wasn't sure if you planned to make more edits in response to copilot comments or just resolve those - so I was waiting to finish review.

File: mica/archive/aca_hdr3.py [hunk 4]
```diff
@@ -48,15 +88,97 @@ def func(slot_data) -> np.ndarray:
...
+    Parameters
+    ----------
+    bytes_list : list of np.ndarray[np.uint8]
+        List of byte arrays representing the quadrant offset values.
+
+    Returns
+    -------
+    np.ndarray[np.float64]
+        Array of computed quadrant offset values.
+    """
```
[the remark this answers is not in the document]
Comment (taldcroft) [reply mica/archive/aca_hdr3.py:142]: Fixed

File: mica/archive/tests/test_aca_hdr3.py [hunk 5]
```diff
@@ -72,3 +81,109 @@ def test_two_byte_sum():
...
+    np.testing.assert_array_equal(idx_lsb, [0, 2])
+
+
+def test_fuzzy_join_times_tolerance_boundary():
+    """Entry at exactly tol is included; entry just over tol is excluded."""
+    msb = np.array([10.0, 20.0, 30.0])
+    lsb = np.array(
+        [10.0, 22.0, 33.0]
+    )  # msb[1]-lsb[1]=2.0 (in), msb[2]-lsb[2]=3.0 (out)
+    idx_lsb, idx_msb = aca_hdr3._fuzzy_join_times(msb, lsb, tol=2.1)
```
[the remark this answers is not in the document]
Comment (taldcroft) [reply mica/archive/tests/test_aca_hdr3.py]: Fixed

**Comment by taldcroft** [comment]
OK, I just pushed another commit which addresses the issues noted by the copilot review. In my defense, copilot generated the test that copilot didn't like!  :smile:.

**Review by taldcroft — COMMENTED** [review]
Still seeing a disconnect using VScode to resolve issues and make comments.

**Review by jeanconn — APPROVED** [review]
